hex/helpers/lib/parse_deps.exs

```diff
@@ -69,6 +69,9 @@ defmodule Parser do
   defp parse_lock({:git, repo_url, checksum, opts}),
     do: {nil, checksum, git_source(repo_url, opts)}
 
+  defp parse_lock({:hex, _app, version, checksum, _managers, _dependencies, _source, _outer_checksum}),
```

Comment:
This checksum is deprecated, the `outer_checksum` should be used instead.

Reply:
To make sure you don't crash again when a new field is added to the lockfile entry you can do something like this:
```
destructure [:hex, _app, version, checksum], Tuple.to_list(lock)
```

Reply:
Ahh should the outer checksum be used in the following line? `do: {version, _outer_checksum, nil}`

How would you write this with the suggested `destructure`?

Reply:
Yes:

```elixir
destructure(
  [:hex, _app, version, _inner_checksum, _managers, _dependencies, _source, outer_checksum],
  Tuple.to_list(lock)
)

{version, outer_checksum, nil}
```

Though I don't know if changing the checksum will cause issues for dependendabot, for example if it will cause conflicts with an internal database of existing checksums.

Reply:
@ericmj cheers! Shouldn't cause any issues for Dependabot as we're not storing any of this info.

I'm still a bit unclear on the exact codes 🤠(don't actually know Elixir), would the method look like so?:

```
defp parse_lock(destructure(
  [:hex, _app, version, _inner_checksum, _managers, _dependencies, _source, outer_checksum],
  Tuple.to_list(lock)
)),
  do: {version, outer_checksum, nil}
```

Would this still would work with the existing method signatures? If you have the time, would love a PR fixing this up 🙇‍♂️

Reply:
It would look like this:

```elixir
defp parse_lock(tuple) when elem(tuple, 0) == :hex do
  destructure(
    [:hex, _app, version, _inner_checksum, _managers, _dependencies, _source, outer_checksum],
    Tuple.to_list(lock)
  )

  {version, outer_checksum, nil}
end
```

And it would remove the need for the other `parse_lock({:hex, ...` clauses.

I can send a PR.